server/src/http/refresher/delta_refresher.rs

```diff
@@ -31,6 +34,32 @@ impl FeatureRefresher {
 
         let key = cache_key(refresh_token);
         self.features_cache.apply_delta(key.clone(), &delta);
+
+        if let Some(mut entry) = self.delta_cache.get_mut(&key) {
+            entry.add_events(&delta.events);
+        } else if let Some(DeltaEvent::Hydration {
+            event_id,
+            features,
+            segments,
+        }) = delta.events.clone().into_iter().next()
+        {
+            self.delta_cache.insert(
+                key.clone(),
+                DeltaCache::new(
+                    DeltaHydrationEvent {
+                        event_id,
+                        features,
+                        segments,
+                    },
+                    100,
+                ),
+            );
+        } else {
+            warn!(
```

Comment:
Can this actually happen? Should this happen if the code is correct? This feels like we've broken some system constraint here

Reply:
No, it the code is correct, it should never happen.